vmc/resource_vmc_cluster.go

```diff
@@ -109,9 +109,9 @@ func resourceClusterCreate(d *schema.ResourceData, m interface{}) error {
 			}
 			d.SetId(clusterID)
 		}
-		if *task.Status == "FAILED" {
-			return resource.NonRetryableError(fmt.Errorf("task failed to create cluster"))
-		} else if *task.Status != "FINISHED" {
+		if *task.Status == model.Task_STATUS_FAILED {
+			return resource.NonRetryableError(fmt.Errorf("task failed to create cluster: %s", *task.ErrorMessage))
```

Comment:
maybe overkill, but it seems we could add some helper methods so that we only use:
```
return nonRetryableErrorForTask(task, "create cluster")
```
or
```
return task.nonRetryableError("create cluster")
```

And the `nonRetryableError` can add the "task failed to" and the task error message.

Reply:
There are all these patterns of code repeating across the different resource types that:
Check if the taks polling client is unauthenticated,
Check for task failure,
Check for task finish

I plan on extracting that code in a separate method to dedup it.